Gem/Code/Source/Components/ScriptableDecalComponent.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Components/ScriptableDecalComponent.h>
+
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/chrono/chrono.h>
+
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/Material/Material.h>
+
+namespace MultiplayerSample
+{
+    void ScriptableDecalComponent::Reflect(AZ::ReflectContext* context)
+    {
+        SpawnDecalConfig::Reflect(context);
+
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<MultiplayerSample::ScriptableDecalComponent, AZ::Component>()
+                ->Version(0)
+                ;
+
+            AZ::EditContext* editContext = serialize->GetEditContext();
+            if (editContext)
+            {
+                editContext->Class<MultiplayerSample::ScriptableDecalComponent>(
+                    "Scriptable Decals", "Allows spawning decals directly from script without prefabs.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Graphics")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZStd::vector<AZ::Crc32>({ AZ_CRC_CE("Level") }))
+                    ;
+            }
+        }
+
+        AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context);
+        if (behaviorContext)
+        {
+            behaviorContext->EBus<DecalRequestBus>("RequestBus")
```

Comment:
Shouldn't this be named "DecalRequestBus"?

Reply:
It must. Otherwise would appear in scripting as `RequestBus`.